### PR TITLE
[SYCLomatic] Fix invalid file path in compilation database when using intercept-build to parse build log file

### PR DESCRIPTION
**Pull request by tomflinda**
Signed-off-by: chenwei.sun <chenwei.sun@intel.com>

### CODE REVIEW

File: clang/tools/scan-build-py/lib/libscanbuild/arguments.py [hunk 1]
```diff
@@ -469,13 +469,13 @@ def parser_build_log(parser):
...
+        help="""Specifies the build log file path of project specified by --work-directory.""")
 
-def parser_in_root(parser):
+def parser_work_directory(parser):
     parser.add_argument(
-        '--in-root',
+        '--work-directory',
         metavar='<path>',
-        help="""Specifies the directory path for the root of the source tree that needs to be migrated.""")
+        help="""Specifies the work directory of the source tree that needs to be migrated.""")
```
Comment (ziranzha) [inline clang/tools/scan-build-py/lib/libscanbuild/arguments.py]: how about use "Specifies the work directory of the build log."

File: clang/tools/scan-build-py/lib/libscanbuild/arguments.py [hunk 2]
```diff
@@ -469,13 +469,13 @@ def parser_build_log(parser):
     parser.add_argument(
         '--parse-build-log',
         metavar='<file>',
-        help="""Specifies the build log file path of project specified by --in-root.""")
+        help="""Specifies the build log file path of project specified by --work-directory.""")
```
Comment (ziranzha) [inline clang/tools/scan-build-py/lib/libscanbuild/arguments.py]: how about use "Specifies the file path of the build log."

File: clang/tools/scan-build-py/lib/libscanbuild/arguments.py (hunk 1 above)
Comment (tomflinda) [reply clang/tools/scan-build-py/lib/libscanbuild/arguments.py]: work directory of source file in build log may be changed, I advice"Specifies the work directory of the source file in the build log that needs to be migrated."

File: clang/tools/scan-build-py/lib/libscanbuild/arguments.py (hunk 2 above)
Comment (tomflinda) [reply clang/tools/scan-build-py/lib/libscanbuild/arguments.py]: OKay

**Review by ziranzha — APPROVED** [review]
LGTM

File: clang/test/dpct/parse_build_log/compile_commands.json_ref [hunk 3]
```diff
@@ -0,0 +1,12 @@
+// CHECK:[
+// CHECK:    {
+// CHECK:        "command": "nvcc -c -I../../common/inc -m64 --std=c++11 -o bar.o -D__CUDACC__=1 bar.cu",
+// CHECK:        "directory": "{{.*}}/parse_build_log/Output",
+// CHECK:        "file": "{{.*}}/parse_build_log/Output/bar.cu"
```
Comment (zhimingwang36) [inline clang/test/dpct/parse_build_log/compile_commands.json_ref:5]: Line 5,  the bar.cu is in parse_build_log folder, so the path here is wrong. 
Will the scripts file check if  " "{{.*}}/parse_build_log/Output/bar.cu" is exist to hint whether -work-dir is set right.

File: clang/tools/scan-build-py/lib/libscanbuild/arguments.py [hunk 4]
```diff
@@ -38,20 +38,20 @@ def parse_args_for_intercept_build():
...
-    if args.parse_build_log and not args.in_root:
-        print('Error: option --in-root needs to be specified, if --build-log is specified')
+    if args.parse_build_log and not args.work_directory:
+        print('Error: option --work-directory needs to be specified, if --build-log is specified')
         sys.exit(-1)
 
-    if not args.parse_build_log and args.in_root:
-        print('Error: option --build-log needs to be specified, if –in-root is specified')
+    if not args.parse_build_log and args.work_directory:
+        print('Error: option --build-log needs to be specified, if -work-directory is specified')
```
Comment (zhimingwang36) [inline clang/tools/scan-build-py/lib/libscanbuild/arguments.py]: -work-directory should be given a default value for user.   consider to use the folder of build log file.

File: clang/test/dpct/parse_build_log/compile_commands.json_ref (hunk 3 above)
Comment (tomflinda) [reply clang/test/dpct/parse_build_log/compile_commands.json_ref:5]: Okay

File: clang/tools/scan-build-py/lib/libscanbuild/arguments.py (hunk 4 above)
Comment (tomflinda) [reply clang/tools/scan-build-py/lib/libscanbuild/arguments.py]: ok

**Review by zhiweij1 — APPROVED** [review]

File: clang/tools/scan-build-py/lib/libscanbuild/arguments.py [hunk 5]
```diff
@@ -469,13 +461,15 @@ def parser_build_log(parser):
...
 
-def parser_in_root(parser):
+def parser_work_directory(parser):
     parser.add_argument(
-        '--in-root',
+        '--work-directory',
         metavar='<path>',
-        help="""Specifies the directory path for the root of the source tree that needs to be migrated.""")
+        default=argparse.SUPPRESS,
+        help="""Specifies the work directory of the source file in the build log that needs to be migrated.
```
Comment (zhimingwang36) [inline clang/tools/scan-build-py/lib/libscanbuild/arguments.py]: how about?  (current working-directory is misleading). 
 -S <path-to-source>          = Explicitly specify a source directory.

File: clang/tools/scan-build-py/lib/libscanbuild/arguments.py [hunk 6]
```diff
@@ -469,13 +461,15 @@ def parser_build_log(parser):
...
-def parser_in_root(parser):
+def parser_work_directory(parser):
     parser.add_argument(
-        '--in-root',
+        '--work-directory',
         metavar='<path>',
-        help="""Specifies the directory path for the root of the source tree that needs to be migrated.""")
+        default=argparse.SUPPRESS,
+        help="""Specifies the working directory of the command that generates the build log.
+                 (default: the directory of build log file, if work directory is not provided)""")
```
Comment (zhimingwang36) [inline clang/tools/scan-build-py/lib/libscanbuild/arguments.py]: Specifies the working directory of the command that generates the build log specified by option --parse-build-log.   Default: the directory of build log file specified by option --parse-build-log.

File: clang/tools/scan-build-py/lib/libscanbuild/arguments.py (hunk 5 above)
Comment (tomflinda) [reply clang/tools/scan-build-py/lib/libscanbuild/arguments.py]: Have synced with Zhiming and Ziran, path-to-source is not accurate, so work-directory is preferred to use.

File: clang/tools/scan-build-py/lib/libscanbuild/arguments.py (hunk 6 above)
Comment (tomflinda) [reply clang/tools/scan-build-py/lib/libscanbuild/arguments.py]: done

**Pull request merged**